docs/generators/MSSqlDDL.md

```diff
@@ -0,0 +1,494 @@
+SET
+  ID = (SELECT MAX(ID) FROM ExampleSchema.Example)+1
+, USER_ID = USER_NAME()
+, TMSTAMP = GETDATE()
+WHERE ID = (SELECT ID FROM INSERTED)
+GO
+```
+
+
+# Audit Trigger
+An Audit trigger can be generated to track `INSERT`, `DELETE` and `UPDATE` queries performed on the table. For this to happed the flag `--flag "audit triggers"` should be passed to the generator. If the flag is passed a secondry table will be generated in the DDL called `<SCHEMA><TABLENAME>Audit`. When an `INSERT`, `UPDATE` or `DELETE` query is executed this table will be inserted into with:
+```
+ACTION
+GETDATE()
+<TABLE FIELDS>
+```
+Example SI file:
+```
+TABLE Example
+    ID                  SEQUENCE
+    SOME_DATA           CHAR(100)
+    USER_ID             USERSTAMP
+    TMSTAMP             TIMESTAMP
+```
+Jportal Params:
+```
+--builtin-generator MSSqlDDL:E:\temp --flag "audit triggers"
+```
+This will generate the DDL:
+```
+CREATE TABLE ExampleSchema.Example
+(
+  ID INTEGER NOT NULL
+, SOME_DATA VARCHAR(100) NOT NULL
+, USER_ID VARCHAR(50) NOT NULL
+, TMSTAMP DATETIME NOT NULL
+, CONSTRAINT PK_ExampleSchema_Example_PKEY PRIMARY KEY (
+    ID
+  )
+)
+GO
+
+IF OBJECT_ID('ExampleSchema.ExampleUpdateTrigger','TR') IS NOT NULL
+    DROP TRIGGER ExampleSchema.ExampleUpdateTrigger
+GO
+
+CREATE TRIGGER ExampleSchema.ExampleUpdateTrigger ON ExampleSchema.Example FOR UPDATE AS
+UPDATE ExampleSchema.Example
+SET
+  USER_ID = USER_NAME()
+, TMSTAMP = GETDATE()
+FROM INSERTED I
+WHERE ExampleSchema.ExampleID = I.ID 
+GO
+
+IF OBJECT_ID('ExampleSchemaExampleAudit','U') IS NOT NULL
+    DROP TABLE ExampleSchemaExampleAudit
+GO
+
+CREATE TABLE ExampleSchemaExampleAudit
+(
+  AuditId INTEGER IDENTITY(1,1) NOT NULL PRIMARY KEY
+, AuditAction INTEGER NOT NULL -- 1 = INSERT, 2 = DELETE, 3 = UPDATE
+, AuditWhen DATETIME NOT NULL
+, ID INTEGER NULL
+, SOME_DATA VARCHAR(100) NULL
+, USER_ID VARCHAR(50) NULL
+, TMSTAMP DATETIME NULL
+)
+GO
+
+IF OBJECT_ID('ExampleSchemaExampleAuditTrigger','TR') IS NOT NULL
+    DROP TRIGGER ExampleSchemaExampleAuditTrigger
+GO
+
+CREATE TRIGGER ExampleSchemaExampleAuditTrigger ON ExampleSchemaExample
+FOR INSERT, DELETE, UPDATE AS
+BEGIN
+  DECLARE @INSERT INT, @DELETE INT, @ACTION INT;
+  SELECT @INSERT = COUNT(*) FROM INSERTED;
+  SELECT @DELETE = COUNT(*) FROM DELETED;
+  IF @INSERT > 0 SELECT @ACTION = 1 ELSE SELECT @ACTION = 0;
+  IF @DELETE > 0 SELECT @ACTION = @ACTION + 2;
+  -- 1 = INSERT, 2 = DELETE, 3 = UPDATE
+  IF @ACTION = 2 BEGIN
+    INSERT INTO ExampleSchemaExampleAudit
+    SELECT @ACTION
+         , GETDATE()
+          , ID
+          , SOME_DATA
+          , USER_ID
+          , TMSTAMP
+    FROM DELETED;
+  END ELSE
+  BEGIN
+    INSERT INTO ExampleSchemaExampleAudit
+    SELECT @ACTION
+         , GETDATE()
+         , ID
+         , SOME_DATA
+         , USER_ID
+         , TMSTAMP
+    FROM INSERTED;
+  END
+END
+GO
+```
+
+
+## Index Information
+Indexes can be created using the `KEY` keyword in the SI file:
```

Comment:
Consider expanding the explanation to explain the syntax as 
KEY <IndexName> [PRIMARY] <Field1> [Field2]...[FieldN]
with a light explanation

Reply:
Agree.

Reply:
A lot of what is in this doc (GRANT, VIEW etc) falls in that category. Will you change it?

Reply:
#67